File: golang/1-twoSum.go
package golang

func TwoSum(nums []int, target int) []int {
	var results []int
	for i, n := range nums {
		matched := target - n
		for j, m := range nums[i:] {
			if m == matched && i != j+i {
				results = []int{i, j + i}
				break
			}
		}
		if len(results) > 0 {
			break
		}
	}
	return results
}

File: ts/1-twoSum.ts
const twoSum = (nums: number[], target: number): number[] => {
  const hash: {
    [key: number]: number
  } = {}
  const result: number[] = []
  nums.some((v, i) => {
    const value = hash[target - nums[i]]
    if (value > -1) {
      result.push(i)
      result.push(value)
    } else {
      hash[v] = i
    }
    return result.length !== 0
  })
  return result
}

export { twoSum }
